server/src/main/java/org/elasticsearch/index/mapper/TimeSeriesParams.java

```diff
@@ -23,20 +23,46 @@ public final class TimeSeriesParams {
 
     private TimeSeriesParams() {}
 
+    /**
+     * There are various types of metric used in time-series aggregations and downsampling.
+     * Each supports different field types and different calculations.
+     * Two of these, the COUNTER and GAUGE apply to most numerical types, mapping a single number,
+     * while the POSITION metric applies only to geo_point and therefor a pair of numbers (lat,lon).
+     * To simplify code that depends on this difference, we use the parameter `scalar==true` for
+     * single number metrics, and `false` for the POSITION metric.
+     */
     public enum MetricType {
         GAUGE(new String[] { "max", "min", "value_count", "sum" }),
-        COUNTER(new String[] { "last_value" });
+        COUNTER(new String[] { "last_value" }),
+        POSITION(new String[] {}, false);
 
         private final String[] supportedAggs;
+        private final boolean scalar;
 
         MetricType(String[] supportedAggs) {
+            this(supportedAggs, true);
+        }
+
+        MetricType(String[] supportedAggs, boolean scalar) {
             this.supportedAggs = supportedAggs;
+            this.scalar = scalar;
         }
 
+        /** list of aggregations supported for downsampling this metric */
         public String[] supportedAggs() {
             return supportedAggs;
         }
 
+        /** an array of metrics representing simple numerical values, like GAUGE and COUNTER */
+        public static MetricType[] scalar() {
+            Object[] objs = Arrays.stream(MetricType.values()).filter(m -> m.scalar).toArray();
```

Comment:
I wonder if we can simplify this function and avoid casting (I am very picky with casting values). Something like:

```
return Arrays.stream(MetricType.values()).filter(m -> m.scalar).toList().toArray(MetricType[]::new);
```

Reply:
Excellent idea. I pushed a fix.